include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -508,8 +509,86 @@ using __repacked_tuple_t = typename __repacked_tuple<T>::type;
 template <typename _ContainerOrIterable>
 using __value_t = typename __internal::__memobj_traits<_ContainerOrIterable>::value_type;
 
-//A contract for future class: <sycl::event or other event, a value or sycl::buffers...>
-//Impl details: inheretance (private) instead of aggregation for enabling the empty base optimization.
+template <typename _T>
+struct __accessor
+{
+  private:
+    using __accessor_t = sycl::accessor<_T, 1, sycl::access::mode::read_write, __dpl_sycl::__target_device,
+                                        sycl::access::placeholder::false_t>;
+    __accessor_t m_acc;
+    _T* m_ptr = nullptr;
+    bool m_usm = false;
+
+  public:
+// Compilers before 2023.1 don't provide a default constructor for sycl::accessor. Fallback to a buffer instead.
+#if __SYCL_COMPILER_VERSION == 20230320 || __SYCL_COMPILER_VERSION >= 20230820
```

Comment:
Does it make sense to introduce a "feature presence" macro in the config header?